pkg/services/object/acl/acl.go

```diff
@@ -459,8 +460,12 @@ func (b Service) findRequestInfo(
 		return info, ErrUnknownRole
 	}
 
-	// find verb from token if it is present
-	verb := sourceVerbOfRequest(req, op)
+	// Check if session token verb is equal to the current operation.
+	// Unknown means everything is allowed.
+	verb := sourceVerbOfRequest(req.token, op)
+	if verb != op {
```

Comment:
Unfortunately, we can't simply compare op and value from token. One request type can produce requests of other type. During the tests, I logged such errors:
```
token: DELETE  op: PUT
token: DELETE  op: HEAD
token: GETRANGEHASH  op: GETRANGE
```

We also have this table in [spec](https://nspcc.ru/upload/neofs-spec-latest.pdf#page=28) of produced request types.
| Base/Gen | PUT | DELETE | HEAD | RANGE | GET | HASH | SEARCH |
|----------|:---:|:------:|:----:|:-----:|:---:|:----:|:------:|
| PUT      | +   | -      | -    | -     | -   | -    | -      |
| DELETE   | +   | -      | +    | -     | -   | -    | +      |
| HEAD     | -   | -      | +    | -     | -   | -    | -      |
| RANGE    | -   | -      | +    | +     | -   | -    | -      |
| GET      | -   | -      | +    | -     | +   | -    | -      |
| HASH     | -   | -      | +    | +     | -   | -    | -      |
| SEARCH   | -   | -      | -    | -     | -   | -    | +      |

Reply:
Fixed.